upb/env.c

```diff
@@ -224,7 +226,9 @@ UPB_FORCEINLINE static void *seeded_alloc(void *ud, void *ptr, size_t oldsize,
   } else {
     // Slow path: fallback to other allocator.
     a->need_cleanup = true;
-    return a->alloc(a->alloc_ud, ptr, oldsize, size);
+    // N.B.: do not pass initial block, since it was user-provided; otherwise,
```

Comment:
Ah, actually I think maybe the correct predicate here is:

```
char *chptr = ptr;
void *from = (chptr >= a->mem_base && chptr < a->mem_limit) ? NULL : ptr;
```

Because we ought to pass ptr in the case that it is *not* from the initial block.